praw/models/reddit/subreddit.py

```diff
@@ -879,7 +915,19 @@ def submit(
             if value is not None:
                 data[key] = value
         if selftext is not None:
-            data.update(kind="self", text=selftext)
+            data.update(kind="self")
+            if inline_media:
+                text_media = {}
+                if isinstance(inline_media, InlineMedia):
+                    inline_media = [inline_media]
+                for image in inline_media:
+                    image._upload(self)
```

Comment:
If `._upload()` is going to be a "private" (underscore-prefixed) method, we shouldn't call it outside of that class.

Plus, the `_upload` method calls right back into a private method in the `Subreddit` class.

I think all of the upload and validation logic should probably be moved into (private methods in) the `Subreddit` class, because ultimately the functionality already lives there, and it's only called from that class.

Reply:
See my changes